cdap-common/src/main/java/co/cask/cdap/common/discovery/ResolvingDiscoverable.java

```diff
@@ -59,4 +59,9 @@ private InetSocketAddress resolve(InetSocketAddress bindAddress) {
     }
     return bindAddress;
   }
+
+  @Override
+  public String toString() {
+    return "ResolvingDiscoverable(" + String.valueOf(discoverable) + ")";
```

Comment:
Also, `Discoverable` is an interface, which a lot of time we just use anonymous inner class to implement them, hence the `toString` of that would be meaningless. I think it's better to have a method to turn a `Discoverable` to string by get the name and address from it.